File: LeetCode-Java/Maximum Depth of Binary Tree.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution 
{
    public int maxDepth(TreeNode root) 
    {
        return root == null ? 0 : max(maxDepth(root.left) + 1, maxDepth(root.right) + 1);
    }
    int max(int left,int right)
    {
        return left > right ? left : right;
    }
}

File: Leetcode-cpp/MaximumDepthOfBinaryTree.cpp
/**
 *
 *  * Definition for binary tree
 *
 *   * struct TreeNode {
 *
 *    *     int val;
 *
 *     *     TreeNode *left;
 *
 *      *     TreeNode *right;
 *
 *       *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 *
 *        * };
 *
 *         */
class Solution {
    public:
        int maxDepth(TreeNode *root) {
            if(root == NULL) return 0;
            int left = maxDepth(root->left),right = maxDepth(root->right);
            return (left > right ? left : right) + 1;
        }
};
